app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::ReservationsController < ApplicationController
+  # GET /reservations
+  def index
+    @reservations = Reservation.includes(:room).where('user_id = ?', current_user)
+
+    render json: @reservations
+  end
+
+  # POST /rooms/1/rservation
+  def create
+    @reservation = Reservation.new(reservation_params)
+    @reservation.update(user: current_user, room: Room.find(params[:room_id]))
+
+    if @reservation.save
+      render json: @reservation, status: :created
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /reservation/1
+  def destroy
+    @reservation = Reservation.find(params[:id])
```

Comment:
[OPTIONAL] `before_action` may make for better reading here by adding `before_action` method before `index` like so:
`before_action :set_reservation, only: [:destroy]`
and define the private method like so:
`def set_reservation
    @reservation = Reservation.find(params[:id])
  end
`

and it will allow you to add more methods if needed in future as well by adding the method in the `only` array.

Reply:
For example, if you added an `update` method